streams/src/main/java/org/apache/kafka/streams/StreamsConfig.java

```diff
@@ -138,6 +138,7 @@
 
     /** {@code client.id} */
     public static final String CLIENT_ID_CONFIG = CommonClientConfigs.CLIENT_ID_CONFIG;
+    private static final String CLIENT_ID_DOC = "An id string, which will be used as a prefix for internal consumer, producer and restore-consumer";
```

Comment:
Thanks for the patch. I guess, we can be a little bit more precise here. It's a prefix for the client id of the internal clients. Might also be good to add the pattern that is used for the id used for the internal clients.